src/compiler/moduleNameResolver.ts

```diff
@@ -82,15 +82,23 @@ interface PathAndExtension {
 
 /**
  * Kinds of file that we are currently looking for.
- * Typically there is one pass with Extensions.TypeScript, then a second pass with Extensions.JavaScript.
  */
-enum Extensions {
-    TypeScript, /** '.ts', '.tsx', or '.d.ts' */
-    JavaScript, /** '.js' or '.jsx' */
-    Json,       /** '.json' */
-    TSConfig,   /** '.json' with `tsconfig` used instead of `index` */
-    DtsOnly,    /** Only '.d.ts' */
-    TsOnly,     /** '.[cm]tsx?' but not .d.ts variants */
+const enum Extensions {
+    TypeScript  = 1 << 0, // '.ts', '.tsx', '.mts', '.cts'
+    JavaScript  = 1 << 1, // '.js', '.jsx', '.mjs', '.cjs'
+    Declaration = 1 << 2, // '.d.ts', etc.
+    Json        = 1 << 3, // '.json'
+
+    ImplementationFiles = TypeScript | JavaScript,
+}
+
+function formatExtensions(extensions: Extensions) {

```

Comment:
FWIW you can get a pretty similar result with `Debug.formatEnum`. Post-modules that's something I want to yank out into a module for use outside of Debug.

Reply:
I didn’t use it mostly because it was in `Debug` and this production code... I know it would work without any issue but it felt wrong.

Reply:
Yeah, I wouldn't have mentioned it but this seemed like it was just in tracing which straddles the line a little.